storage/range.go

```diff
@@ -1753,7 +1765,25 @@ func (r *Range) AdminMerge(args *proto.AdminMergeRequest, reply *proto.AdminMerg
 }
 
 // InternalChangeReplicas is called after a raft configuration change has committed.
-func (r *Range) InternalChangeReplicas(req *proto.InternalChangeReplicasRequest,
-	resp *proto.InternalChangeReplicasResponse) {
-	// TODO(bdarnell): apply the change to r.Desc.
+func (r *Range) InternalChangeReplicas(batch engine.Engine,
+	req *proto.InternalChangeReplicasRequest,
+	reply *proto.InternalChangeReplicasResponse) {
+	r.Lock()
+	defer r.Unlock()
+	// Apply the committed membership change to r.Desc.
+	// TODO(bdarnell): does this need special atomicity consideration, or will it be
```

Comment:
What we do need a TODO for is updating the meta record(s) for the range with the new replica info.

It occurs to me that we might need to rework what we're doing here and instead make changing replicas part of a distributed txn.

The other mechanism which would work is to do exactly this, but set a dirty bit in the range descriptor and have the leader of the range update the meta transaction records asynchronously. It's fine if they lag a bit as clients are caching their values anyway and don't depend on them being fully up to date for correctness. Clients with stale info should get turned away from any range that they contact which ends up no longer being a member of the group.